mslib/mscolab/server.py

```diff
@@ -642,6 +643,56 @@ def import_permissions():
                     "message": message})
 
 
+@APP.route('/reset_password/<token>', methods=['GET', 'POST'])
+def reset_password(token):
+    try:
+        email = confirm_token(token)
+    except TypeError:
+        return jsonify({"success": False}), 401
+    if email is False:
+        return jsonify({"success": False}), 401
+    user = User.query.filter_by(emailid=email).first_or_404()
+    if user.confirmed:
+        form = ResetPasswordForm()
+        if form.validate_on_submit():
+            try:
+                user.hash_password(form.confirm_password.data)
+                db.session.commit()
+                flash('Password reset Success. Please login by the user interface.', 'category_success')
+                return render_template('user/status.html')
+            except IOError:
+                flash('Password reset failed. Please try again later', 'category_danger')
+        return render_template('user/reset_password.html', form=form)
+    else:
+        return jsonify({"success": False}), 401
+
+
+@APP.route("/reset_request", methods=['GET', 'POST'])
+def reset_request():
+    form = ResetRequestForm()
+    if form.validate_on_submit():
+        # Check wheather user exists or not based on the db
+        user = User.query.filter_by(emailid=form.email.data).first()
+        if user:
+            try:
+                username = user.username
+                token = generate_confirmation_token(form.email.data)
+                reset_password_url = url_for('reset_password', token=token, _external=True)
```

Comment:
We should have a timeout that this token is  not endless available, e.g. can we have a lifetime of 24h

Reply:
We had previously set the expiration time to 1 hour using the `confirm_token` method. However, I have updated it to 24 hours.